Scripts/script-MatchRegex.yml

```diff
@@ -6,37 +6,51 @@ script: |-
   var data = (typeof args.data) === 'string' ? args.data : JSON.stringify(args.data);
   var flags = args.flags ? args.flags : 'im';
   var r = new RegExp(args.regex, flags);
-  var m = r.exec(data);
-  if (m && m.length > 0) {
-      var val = m[0];
-      var group = parseInt(args.group);
-      if (args.group && m.length > group) {
-          val = m[group];
+  var m;
+  var vals = [];
+  while ((m = r.exec(data)) !== null) {
+      if (m && m.length > 0) {
+          var val = m[0];
+          var group = parseInt(args.group);
+          if (args.group && m.length > group) {
+              val = m[group];
+          }
+          var ec = {};
+          if (args.contextKey) {
+              ec[args.contextKey] = val;
+          }
       }
-      var ec = {};
-      if (args.contextKey) {
-          ec[args.contextKey] = val;
-      }
-      return {Type: entryTypes.note, Contents: val, ContentsFormat: formats.text, EntryContext: ec};
   }
+  if (vals.length) {
```

Comment:
val is never added into vals, causing vals to always be empty.

Reply:
Whoops, I had some dedup code that I realized wouldn't be ideal for every scenario, that's where I was adding into vals before I got rid of it.